File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        writer = 0 
        
        for scout in range(1, len(nums)):
            # If the scout finds a new number
            if nums[scout] != nums[writer]:
                writer += 1        # Move writer to the next empty slot
                nums[writer] = nums[scout] # Write the new number there
        
        return writer + 1

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        set<int> original;
        for(int i = 0;i<nums.size();i++){
            original.insert(nums[i]);
        }

        int index = 0;

        for(auto i : original){
            nums[index] = i;
            index++;
        }

        return original.size();
    }
};